sample-apps/go-sample-app/collection/client.go

```diff
@@ -59,59 +55,73 @@ var traceCommonLabels = []attribute.KeyValue{
 	attribute.String("port", cfg.Port),
 }
 
-// StartClient starts the OTEL controller which periodically collects signals and exports them.
+// StartClient starts the traces and metrics providers which periodically collects signals and exports them.
 // Trace exporter and Metric exporter are both configured.
 func StartClient(ctx context.Context) (func(context.Context) error, error) {
 
+	if id, present := os.LookupEnv("INSTANCE_ID"); present {
+		testingId = "_" + id
+	}
+	res := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.ServiceName("go-sample-app"),
+	)
+	if _, present := os.LookupEnv("OTEL_RESOURCE_ATTRIBUTES"); present {
+		envResource, err := resource.New(ctx, resource.WithFromEnv())
+		if err != nil {
+			return nil, err
+		}
+		res = envResource
+	}
+
 	// Setup trace related
-	tp, err := setupTraceProvider(ctx)
+	tp, err := setupTraceProvider(ctx, res)
 	if err != nil {
 		return nil, err
 	}
 
 	otel.SetTracerProvider(tp)
 	otel.SetTextMapPropagator(xray.Propagator{}) // Set AWS X-Ray propagator
 
-	// Setup metric related
-	ctrl, err := setupMetricsController(ctx)
+	exp, err := otlpmetricgrpc.New(ctx, otlpmetricgrpc.WithInsecure())
 	if err != nil {
 		return nil, err
 	}
-	global.SetMeterProvider(ctrl)
+	meterProvider := metric.NewMeterProvider(metric.WithResource(res), metric.WithReader(metric.NewPeriodicReader(exp)))
+
+	otel.SetMeterProvider(meterProvider)
 
 	return func(context.Context) (err error) {
```

Comment:
nit: Can we avoid using a named return value here? It's not used consistently through the program and could be confusing. It could be easier to use `return err` in line.

Reply:
Avoided using a named return value here in my latest commit.